sequencer/src/api/sql.rs

```diff
@@ -159,6 +159,35 @@ impl CatchupStorage for SqlStorage {
         ))?;
         load_chain_config(&mut tx, commitment).await
     }
+
+    async fn get_leaf_chain(&self, height: u64) -> anyhow::Result<Vec<Leaf2>> {
+        let mut tx = self
+            .read()
+            .await
+            .context(format!("opening transaction to fetch leaf at {height}"))?;
+        let leaf = tx
+            .get_leaf((height as usize).into())
+            .await
+            .context(format!("leaf {height} not available"))?;
+        let last_leaf: Leaf2 = leaf.leaf().clone().into();
+        let mut chain = vec![last_leaf.clone()];
+        let mut h = height + 1;
+        while let Ok(leaf) = tx.get_leaf((h as usize).into()).await {
+            let leaf: Leaf2 = leaf.leaf().clone().into();
+            chain.push(leaf.clone());
+            // just one away from deciding
+            if leaf.view_number() == last_leaf.view_number() + 1 {
```

Comment:
The query service only has decided leaves in storage, so wouldn't this check always pass? Also, let's say this fails.

Suppose we insert Leaf (bh = 100) into the chain before this loop. If in this loop, leaf (bh= 104) is inserted, then how is this a valid chain?

Reply:
We can use the `get_leaf_range()` if we want the h, h+1, h+2 chain.

`tx.get_leaf_range(height as usize..=(height as usize + 2)).await`